engine/server/sv_main.c

```diff
@@ -153,6 +153,69 @@ qboolean SV_HasActivePlayers( void )
 	return false;
 }
 
+/*
+================
+SV_GetConnectedClientsCount
+
+returns connected clients count (and optionally bots count)
+================
+*/
+int SV_GetConnectedClientsCount(int *bots)
+{
+	int index;
+	int	clients;
+
+	clients = 0;
+	if( svs.clients )
+	{
+		if( bots )
+			*bots = 0;
+
+		for( index = 0; index < svs.maxclients; index++ )
+		{
+			if( svs.clients[index].state >= cs_connected )
+			{
+				if( FBitSet( svs.clients[index].flags, FCL_FAKECLIENT ))
+				{
+					if( bots )
+						*bots++;
+				}
+				else
+					clients++;
+			}
+		}
+	}
+	return clients;
+}
+
+/*
+================
+SV_UpdateStatusLine
+
+updates dedicated server status line in console
+================
+*/
+void SV_UpdateStatusLine( void )
```

Comment:
If this function doesn't do anything on other platforms, it makes sense to move it to `platform/win32` and rename to Platform_UpdateStatusLine. And leave it empty for other, so no cycles wasted on Linux or Android.

Reply:
```
#if XASH_WIN32
void Platform_UpdateStatusLine( void );
#else 
static inline void Platform_UpdateStatusLine( void ) { }
#endif      
```

Just drop it in platform.h, and it's ok

Reply:
I think it just should be implemented for other platforms too in future. Something like it works in ReHLDS. I would implement it also for Linux but I don't know how to work with it :)

Reply:
It's an out of scope of this PR. Anyway, it's gonna be very platform-specific. On Android I would like to put that in notification bar, on Linux... we'll see, I'm dreaming about making an ncurses based console. :)

Reply:
Then ok I'll do it platform-specific

Reply:
But there are some issues:
1. Required serverside data unavailable from platform code
2. Can I mix platform-specific and serverside code together? I think it's bad practice.

Reply:
Maybe just don't call this function for other platforms instead?

Reply:
We had this in Old Engine and it was a mess of macros in random places over the whole engine.

There is still work to be done, of course.

Reply:
>Can I mix platform-specific and serverside code together? I think it's bad practice.

Why not, server side code is always available.
It can be that client code isn't available and there is XASH_DEDICATED macro for that.

Reply:
Но даже если это вынести в платформо-специфичную часть, мне все равно придётся входные данные считать снаружи из серверного кода, опять таки это будут лишние циклы, если не юзать #ifdef

Reply:
Вроде понял, как сделать нормально